source/UF/ConnectedTypes.lagda

```diff
@@ -0,0 +1,223 @@
+as surjections.
+
+\begin{code}
+
+ open propositional-truncations-exist pt
+
+ one-connected-is-inhabited : {X : 𝓤 ̇}
+                            → X is 1 connected → ∥ X ∥
+ one-connected-is-inhabited X-1-conn = one-trunc-to-prop-trunc (center X-1-conn)
+
+ inhabited-is-one-connected : {X : 𝓤 ̇}
+                            → ∥ X ∥ → X is 1 connected
+ inhabited-is-one-connected x-anon =
+  pointed-props-are-singletons (prop-trunc-to-one-trunc x-anon) one-trunc-is-prop
+
+ one-connected-iff-inhabited : {X : 𝓤 ̇}
+                             → X is 1 connected
+                             ↔ ∥ X ∥
+ one-connected-iff-inhabited =
+  (one-connected-is-inhabited , inhabited-is-one-connected)
+
+ one-connected-map-is-surj : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y}
+                           → map f is 1 connected
+                           → is-surjection f
+ one-connected-map-is-surj f-1-con y = one-connected-is-inhabited (f-1-con y)
+
+ surj-is-one-connected-map : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y}
+                           → is-surjection f
+                           → map f is 1 connected
+ surj-is-one-connected-map f-is-surj y = inhabited-is-one-connected (f-is-surj y)
+
+ one-connected-map-iff-surj : {X : 𝓤 ̇} {Y : 𝓥 ̇} {f : X → Y}
+                            → map f is 1 connected
+                            ↔ is-surjection f
+ one-connected-map-iff-surj = (one-connected-map-is-surj , surj-is-one-connected-map)
+
+\end{code}
+
+We now devlop some closure conditions pertaining to connectedness. Connectedness
+is closed under equivalence as expected, but more importantly connectedness
+extends below: more precisely if a type is k connected then it is l connected
+for all l ≤ k. We provide many incarnations of this fact below which may prove
+useful. 
+
+\begin{code}
+
+ connectedness-closed-under-equiv : {X : 𝓤 ̇} {Y : 𝓥 ̇} {k : ℕ}
+                                  → X ≃ Y
+                                  → Y is k connected
+                                  → X is k connected
+ connectedness-closed-under-equiv e Y-con =
+   equiv-to-singleton (truncation-closed-under-equiv e) Y-con
+
+ contractible-types-are-connected : {X : 𝓤 ̇} {n : ℕ}
+                                  → is-contr X
+                                  → X is n connected
+ contractible-types-are-connected {𝓤} {X} {n} (c , C) = ((∣ c ∣[ n ]) , C')
+  where
+   C' : (s : ∥ X ∥[ n ]) → (∣ c ∣[ n ]) ＝ s
+   C' = ∥∥ₙ-ind (hlevels-are-closed-under-id ∥∥ₙ-hlevel (∣ c ∣[ n ]))
+                (λ x → ap ∣_∣[ n ] (C x))
+
+ connectedness-is-lower-closed : {X : 𝓤 ̇} {k : ℕ}
+                               → X is (succ k) connected
+                               → X is k connected
+ connectedness-is-lower-closed {𝓤} {X} {k} X-succ-con =
+   equiv-to-singleton successive-truncations-equiv 
+                      (contractible-types-are-connected X-succ-con)
+
+ connectedness-extends-to-zero : {X : 𝓤 ̇} {k : ℕ}
```

Comment:
This lemma is a little strange because _any_ type is `0`-connected, essentially by definition.

Reply:
I'll remove it.

Reply:
I think I thought it was a base case for something, but it didn't end up being needed.